scripts/run_acceptance_tests.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Python execution for running e2e tests."""
+
+from __future__ import annotations
+
+import argparse
+import contextlib
+import os
+import subprocess
+import sys
+
+from typing import Final, List, Optional, Tuple
+
+# TODO(#15567): This can be removed after Literal in utils.py is loaded
+# from typing instead of typing_extensions, this will be possible after
+# we migrate to Python 3.8.
+from scripts import common  # isort:skip pylint: disable=wrong-import-position, unused-import
+
+from core.constants import constants  # isort:skip
+from scripts import build  # isort:skip
+from scripts import flake_checker  # isort:skip
+from scripts import install_third_party_libs  # isort:skip
+from scripts import servers  # isort:skip
+
+MAX_RETRY_COUNT: Final = 3
+GOOGLE_APP_ENGINE_PORT: Final = 8181
+ELASTICSEARCH_SERVER_PORT: Final = 9200
+PORTS_USED_BY_OPPIA_PROCESSES: Final = [
+    GOOGLE_APP_ENGINE_PORT,
+    ELASTICSEARCH_SERVER_PORT,
+]
+
+_PARSER: Final = argparse.ArgumentParser(
+    description="""
+Run this script from the oppia root folder:
+   python -m scripts.run_e2e_tests
+
+The root folder MUST be named 'oppia'.
+
+NOTE: You can replace 'it' with 'fit' or 'describe' with 'fdescribe' to run a
+single test or test suite.
+""")
+
+_PARSER.add_argument(
+    '--skip-install',
+    help='If true, skips installing dependencies. The default value is false.',
+    action='store_true')
+_PARSER.add_argument(
+    '--skip-build',
+    help='If true, skips building files. The default value is false.',
+    action='store_true')
+_PARSER.add_argument(
```

Comment:
Don't include arguments that you're not using. Make sure all the docstrings for args are correct.

Reply:
@ashish-patwal I don't think you've responded to this comment.

Reply:
@ashish-patwal You still haven't responded to this comment.

Reply:
@seanlip I have now removed all unnecessary flags and arguments. I will give you a brief reason why i am keeping anything else. 
`--skip-build` : very much necessary because our CI utilizes the build which is made during the build step in our CI. So we need to have a skip build options for our CI where we already have a build package and we can directly download and unpack the resources from. When we are running our tests locally then we don't need the `--skip-build` flag cause we have to first build our project then only test it. In `dev_mode` the build time is prettery low, but for `prod_env` it's comparitively very high.

`--prod_env` : a flag to indicate whether we want to build our project in dev or prod env. When testing on CI we need to have `--prod_env` bcoz we are going to test it on a production build. But when testing locally we can skip this option.

`--suite` : the name of the suite. self explainatory

`--server_log_level` : I believe this can be pretty useful for debugging. It's a given argument for dev appserver. Thus i believe it is useful to keep it.

`--source_maps` : I am not so sure about using source maps argument because never used it.

Reply:
Thanks, this seems fine. For the name of the suite, I suggest giving an example so that folks know what to enter. 

Also are you supporting the no-argument case? I thought a suite was mandatory -- the script should fail if a suite is not specified. Please test this and also fix the doc for that argument.

Reply:
@ashish-patwal This still has no reply.

Reply:
@seanlip I have applied the changes. the `--suite` has `required=true` which ensures a user must send some value to the suite flag